packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -416,36 +416,129 @@ class DaemonDomain extends Domain {
   /// is correct.
   Future<Map<String, Object>> getSupportedPlatforms(Map<String, Object?> args) async {
     final String? projectRoot = _getStringArg(args, 'projectRoot', required: true);
-    final List<String> result = <String>[];
+    final List<String> platforms = <String>[];
```

Comment:
nit: should we rename some of these variables (that aren't part of the API) `platformType` and/or add a comment to make it a bit clearer that almost all code here is platform types and not platforms?

Reply:
I will change this variable name, but I would add that--in my mind--the situation is far more confusing than just that we use the wrong terminology here.

I don't think the tool has a standardized concept of what exactly a target platform is. It is probably *roughly* something like operating system + device architecture, but it's complicated by for example the web (js vs WASM), the tester device, the preview device, and custom devices. In particular, there is a lot of complexity in the build_info.dart library: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_info.dart#L509

Similarly, even though we have an enum named `PlatformType`, I wouldn't call that a standardized concept in the tool.